service/history/decision/handler.go

```diff
@@ -330,60 +342,53 @@ Update_History_Loop:
 		}
 
 		executionInfo := msBuilder.GetExecutionInfo()
-
-		scheduleID := token.ScheduleID
-		currentDecision, isRunning := msBuilder.GetDecisionInfo(scheduleID)
+		currentDecision, isRunning := msBuilder.GetDecisionInfo(token.ScheduleID)
 
 		// First check to see if cache needs to be refreshed as we could potentially have stale workflow execution in
 		// some extreme cassandra failure cases.
-		if !isRunning && scheduleID >= msBuilder.GetNextEventID() {
-			handler.metricsClient.IncCounter(metrics.HistoryRespondDecisionTaskCompletedScope, metrics.StaleMutableStateCounter)
-			handler.logger.Error("Encounter stale mutable state in RespondDecisionTaskCompleted",
-				tag.WorkflowDomainName(domainEntry.GetInfo().Name),
-				tag.WorkflowID(workflowExecution.GetWorkflowID()),
-				tag.WorkflowRunID(workflowExecution.GetRunID()),
-				tag.WorkflowScheduleID(scheduleID),
-				tag.WorkflowNextEventID(msBuilder.GetNextEventID()),
-			)
+		if !isRunning && token.ScheduleID >= msBuilder.GetNextEventID() {
+			scope.IncCounter(metrics.StaleMutableStateCounter)
+			logger.Error("Encounter stale mutable state in RespondDecisionTaskCompleted", tag.WorkflowNextEventID(msBuilder.GetNextEventID()))
 			// Reload workflow execution history
 			wfContext.Clear()
 			continue Update_History_Loop
 		}
 
-		if !msBuilder.IsWorkflowExecutionRunning() || !isRunning || currentDecision.Attempt != token.ScheduleAttempt ||
-			currentDecision.StartedID == common.EmptyEventID {
+		if !msBuilder.IsWorkflowExecutionRunning() || !isRunning || currentDecision.Attempt != token.ScheduleAttempt || currentDecision.StartedID == common.EmptyEventID {
+			logger.Debugf("Decision task not found. IsWorkflowExecutionRunning: %v, isRunning: %v, currentDecision.Attempt: %v, token.ScheduleAttempt: %v, currentDecision.StartID: %v",
```

Comment:
This will make log matching tricky if you use `debugf`. I would suggest making these tags.

Reply:
Yeah in some cases it's nice to move variables to tags so message can be used to group logs but certain debug logs don't fit in that category and better to include more context inline. We can always get count of such logs via regex